sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/src/Program.cs

```diff
@@ -0,0 +1,202 @@
+            }
+
+            metrics.Client.TrackEvent("Test run is ending.");
+            Console.Out.WriteLine("Test run is ending.");
+        }
+        catch (TaskCanceledException)
+        {
+            // Run is complete
+        }
+        catch (Exception ex) when
+            (ex is OutOfMemoryException
+            || ex is StackOverflowException
+            || ex is ThreadAbortException)
+        {
+            throw;
+        }
+        catch (Exception ex)
+        {
+            metrics.Client.TrackException(ex);
+            Console.Out.WriteLine($"Exception: {ex.StackTrace}");
+        }
+        finally
+        {
+            // We need to wait one minute after flushing the Application Insights client. The Application
+            // Insights flush is non-deterministic, so we don't want to let the application close until
+            // all telemetry has been sent.
+            metrics.Client.Flush();
+            await Task.Delay(60000).ConfigureAwait(false);
+        }
+    }
+
+    /// <summary>
+    ///   Converts a string into a <see cref="TestScenarioName"/> value.
+    /// </summary>
+    ///
+    /// <param name="testScenario">The string to convert to a <see cref="TestScenarioName"/>.</param>
+    ///
+    /// <returns>The <see cref="TestScenarioName"/> of the string input.</returns>
+    ///
+    public static TestScenarioName StringToTestScenario(string testScenario) => testScenario switch
+    {
+        DataMovementBlobStressConstants.TestScenarioNameStr.UploadSingleBlockBlob => TestScenarioName.UploadSingleBlockBlobTest,
+        DataMovementBlobStressConstants.TestScenarioNameStr.UploadDirectoryBlockBlob => TestScenarioName.UploadDirectoryBlockBlobTest,
+        DataMovementBlobStressConstants.TestScenarioNameStr.DownloadSingleBlockBlob => TestScenarioName.DownloadSingleBlockBlobTest,
+        _ => throw new ArgumentNullException(),
+    };
+
+    /// <summary>
+    ///   The available command line options that can be parsed.
+    /// </summary>
+    ///
+    internal class Options
+    {
+        [Option('t', "test", HelpText = "Test scenario to run.")]
+        public string Test { get; set; }
+
+        [Option('s', "size", HelpText = "Size of each objects to transfer.")]
+        public int Size { get; set; }
+
+        [Option('b', "blockSize", HelpText = "Size of the chunk/block size")]
+        public int BlockSize { get; set; }
+
+        [Option('i', "initialTransferSize", HelpText = "Initial transfer size.")]
+        public int InitialTransferSize { get; set; }
+
+        [Option('c', "count", HelpText = "Number of objects to transfer.")]
+        public int Count { get; set; }
+
+        [Option('d', "duration", HelpText = "Duration of the test run.")]
+        public int Duration { get; set; }
+
+        [Option('p', "parallel", HelpText = "Maximum concurrency.")]
+        public int Parallel { get; set; }
```

Comment:
Is this actually user-settable in DMLib?

Reply:
Yes, `TransferManagerOptions.MaximumConcurrency`